sample-app/src/utils/search-operations.ts

```diff
@@ -0,0 +1,73 @@
+import { AnswersActions } from "@yext/answers-headless-react";
+import { SearchIntent } from "@yext/answers-headless";
+
+const defaultGeolocationOptions: PositionOptions = {
+  enableHighAccuracy: false,
+  timeout: 6000,
+  maximumAge: 300000,
+};
+
+/**
+ * If the provided search intents include a 'NEAR_ME' intent, retrieve and
+ * store user's location in headless state. Then, execute a search with
+ * user's location, if that's successfully retrieved, attached to the request.
+ */
+export async function executeSearchWithIntents(
+  answersActions: AnswersActions,
+  isVertical: boolean,
+  intents: SearchIntent[],
+  geolocationOptions?: PositionOptions
+) {
+  if (intents.includes(SearchIntent.NearMe)) {
+    try {
+      const position = await getUserLocation(geolocationOptions);
+      answersActions.setUserLocation({
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+      });
+    } catch(e) {
+      console.error(e);
+    }
+  }
+  executeSearch(answersActions, isVertical);
+}
+
+/**
+ * Executes a universal/vertical search
+ */
+export async function executeSearch(answersActions: AnswersActions, isVertical: boolean) {
+  isVertical
+    ? answersActions.executeVerticalQuery()
+    : answersActions.executeUniversalQuery();
+  return;
+}
+
+/**
+ * Get search intents of the current query stored in headless using autocomplete request.
+ */
+export async function getSearchIntents(answersActions: AnswersActions, isVertical: boolean) {
+  const results = isVertical
+    ? await answersActions.executeVerticalAutocomplete()
+    : await answersActions.executeUniversalAutocomplete();
+  return results?.inputIntents;
+}
+
+/**
+ * Retrieves user's location using nagivator.geolocation API
+ */
+export async function getUserLocation(geolocationOptions?: PositionOptions): Promise<GeolocationPosition> {
```

Comment:
In the SDK, we don't try to get the location again if the location is already in storage. Should we have that behavior here as well? The API can take a while to get a location, so it could slow things down if we use `navigator.geolocation.getCurrentPostion` on every near me query

Reply:
Similarly, if the location is already in storage, we don't need to fire off an additional autocomplete request to check the search intents

Reply:
I see. I think we should leave `getUserLocation` as is, since user may want to manually update their location through the button in `LocationBias` (which uses this function) if their location did change somehow. But I will update `executeSearchWithIntents` to first check the state before calling this function!